p2p/src/authenticated/lookup/actors/listener.rs

```diff
@@ -144,19 +167,223 @@ impl<E: Spawner + Clock + ReasonablyRealtime + Network + Rng + CryptoRng + Metri
                     continue;
                 }
             };
-            debug!(ip = ?address.ip(), port = ?address.port(), "accepted incoming connection");
+            debug!(?address, "accepted incoming connection");
+
+            // Cleanup the rate limiters periodically
+            if accepted > CLEANUP_INTERVAL {
+                self.ip_rate_limiter.shrink_to_fit();
+                self.subnet_rate_limiter.shrink_to_fit();
+                accepted = 0;
+            }
+            accepted += 1;
+
+            // Check whether the IP (and subnet) exceeds its rate limit
```

Comment:
I believe it is most correct to increment the rate limiter for both IP and subnet (rather than exiting early if IP is hit first).